### DISCUSSION TITLE
GoPro Labs add logo to videos via QR code not working

### DISCUSSION THREAD

**Discussion by don-gopro-user22** [gopro/labs#300] [Q&A]
Hi GoPro Labs Community,

I found GoPro Labs Add Logo to Video to be an excellent feature for adding a logo for livestreaming from my ​Hero 8 using a custom QR code. The tutorial is linked below:

https://gopro.github.io/labs/control/logo/

However, after upgrading to a Hero 10​, the same technique simply doesn't work. I place the correctly sized PNG logo file in the MISC folder on the SD card. 

The QR code is then created on the tutorial page when I enter the file name in the given space. The GoPro Hero 10 successfully scans the QR code (beeps) but the logo doesn't appear on the GoPro screen.

​Can anyone suggest a reason why this is the case

GoPro Labs firmware used: 10_01_50_70

Thanks in advance,

Don

**Comment by dnewman-gpsw** [comment]
On HERO10 and 11, the overlay functions have more limitations.  First limitation (documented), only 16:9 resolutions at 4kp30, 2.7k60 or lower work. The second (undocumented), there is no preview of the overlay on the LCD, in only renders into the encoded video.  The second limitation is cosmetic, and is not noticed by anyone accept those coming from HERO8 or 9, and who have used this feature before.  These differences are due to architectural changes between the GP1 and GP2 processors.

**Comment by don-gopro-user22** [comment]
Hi David,

Thank you so much for your quick response!

I have since tested it and can confirm it works with standard video
captured to the micro sd card.

However, I need it to work for livestreamed video.  I have tested this with
the Hero 10 and the logo overlay does not appear in the livestream.

It worked well for livestreams shot using the Hero 8.  Is there a
workaround to get the logo to appear using Hero 10 livestreams?

I would really love there to be a solution for the Hero 10. (fingers
crossed!!!)

Best wishes,

Don
Don Kelly
PetMinder Portlaoise
www.petminder.ie
086 1628297

**Reply by dnewman-gpsw** [reply]
I just confirmed your findings on HERO11.  It seems the overlays are rendering to the captures, but not the Livestream feed.  You are the first to find this.  If this can be fixed on H11, it might be migratable to H10, both should have updated builds latter in the year.

**Comment by don-gopro-user22** [comment]
Great, thanks for the update David. I appreciate you confirming my
findings. It would be awesome if it gets fixed for rhe 10 as it is integral
to my business. I would be grateful if you could let me know if it is
resolved.

Kind regards,
Don

**Reply by dnewman-gpsw** [reply]
Use HERO8 in the meantime.   Even GoPro.com is selling stocks of HERO8 (while supplies last.)

**Comment by don-gopro-user22** [comment]
Will do, thanks!

**Comment by dnewman-gpsw** [comment]
Made some progress on a patch, are willing to experiment with untested Labs firmware?

**Comment by don-gopro-user22** [comment]
Hi David,

Many thanks for your email!

Sure, of course. I'd definitely be willing to give it a try!

Warm regards,

Don

**Reply by dnewman-gpsw** [reply]
Here is a the expiring link https://www.dropbox.com/t/7AJ3xQYnNpzVsu5O (good for 7 days).

This is still 1.50.7x firmware with a tweak to add overlays and logo support on the livestream. The performance may not be the same as your earlier experience, Logo size might need to be smaller, as I haven't done much testing. The example GoPro logo on https://gopro.github.io/labs/control/logo/ works fine.

**Reply by dnewman-gpsw** [reply]
any feedback?

**Comment by don-gopro-user22** [comment]
Hi David,

Many thanks for creating the patch for the logo display issue.  I really
appreciate the time and effort you have put into this.

Unfortunately, I have yet to get it to work.  I have outlined the steps I
have taken below.  Perhaps I have missed an important step which you might
spot?

1. Formatted the micro sd card

2. Factory reset on my GoPro Hero 10

3. Paired device with Quik app

4. Set GoPro to 1080p 30 fps

5.  Install 1.50.7x firmware (Update folder placed in root folder of sd
card)

6. Added GoPro logo from https://gopro.github.io/labs/control/logo/ to MISC
folder of sd card

7. Create QR code from https://gopro.github.io/labs/control/logo/
using GoProLogo.png as filename

8. Scan newly created QR code with Hero 10

9. Confirm logo is displaying with recorded video capture 👍

10. Initiate livestreaming, GoPro logo is not displaying...


I would really appreciate any feedback you can provide on the above.

Warm regards,

Don

Don Kelly
PetMinder Portlaoise
www.petminder.ie
086 1628297

**Reply by dnewman-gpsw** [reply]
I just confirmed with a different HERO10, adding the firmware, checked it was now 1.50.72.

Added this Logo on the MISC folder of SD Card: https://gopro.github.io/labs/control/logo/GoProLogo.png 

Setup the logo with: https://gopro.github.io/labs/control/set/?cmd=oMBRNX=20,20oMBRNP=%22TL%22oMBRNT=0oMLOGO=%22GoProLogo.png%22

I used this page to setup WiFi and RTMP addresses https://gopro.github.io/labs/control/rtmp/
Starting the Livestream with:  https://gopro.github.io/labs/control/set/?cmd=oW1mVr1080!W!GL

Worked as expected:
![image](https://user-images.githubusercontent.com/10856084/231845954-f3f42450-6c46-4f79-92b1-79fe60c18247.png)

I have not trying livestreaming through the App.  Did you do anything different?

**Comment by don-gopro-user22** [comment]
Thanks David,

Really appreciate your help with this.

I'll try it your way and see how it goes.

Best wishes,

Don

**Comment by don-gopro-user22** [comment]
Hi David,

I just got that working!!!

Sincere thanks for all your time and energy that you gave to help me with
this! 👍🏻

You're a star! ⭐ ⭐⭐⭐⭐

Wishing you the best for the future 😁

Warm regards,

Don

**Reply by dnewman-gpsw** [reply]
Any idea why it wasn't working in your first tries?  What did you do differently?

**Comment by don-gopro-user22** [comment]
Hi David,

It just seemed to work because of initiating the livestreaming step via the
QR codes.  Whatever route the Quik app takes in initiating the livestream,
the logo was getting left out.

I'm just delighted it works 👍🏻🙂

Don

**Reply by dnewman-gpsw** [reply]
That is the info I needed to fix that in the official version.

**Comment by don-gopro-user22** [comment]
Hi David,

After testing it in work today, although the logo appears for
livestreaming, the video is frozen. (When I tested it for the first time at
home I didn't move the GoPro camera so it looked like everything was
working fine!).

The background audio works fine, but the video is frozen.

Do you think it is possible to initiate the livestreaming with the logo
present using the Quik app.

Best wishes,

Don

**Reply by dnewman-gpsw** [reply]
What size of the PNG used?  Are using a location encoding as well?  The more work the camera does the more likely this can happen.  The GoPro logo without local storage was stable in my testing.  If your logo is failing, try making it smaller.

**Reply by dnewman-gpsw** [reply]
Is your logo monochrome?  One performance enhancement I considering is to support grayscale only logos, this would allow for larger graphics to be stables rendered.  This is not in the current firmware, just an idea.

**Comment by don-gopro-user22** [comment]
Thanks for your feedback David, much appreciated.

I'll do some more testing and get back to you.

Best wishes,

Dob

**Reply by dnewman-gpsw** [reply]
Any updates on your testing?

**Comment by dnewman-gpsw** [comment]
I have refined the code for more performance: https://www.dropbox.com/t/JaLyAX8WbV7HQif7  (new expiring link, good for another 7 days)

Should be more stable. If you have a B&W png (greyscale,) the performance is optimized further.  

This should also work from the Quik App for Livestreams.

**Comment by don-gopro-user22** [comment]
Hi David,

Thanks for your emails.

I have reduced the size of my logo and so far this appears to have worked.
Originally, my logo dimensions were 400 X 100 px which was recommended in
the documentation and it worked fine on my Hero 8.

The same logo file didn't work with the Hero 10/new labs firmware until I
reduced it to 200 X 50 px.

I will assess it over the coming week in terms of stability and try your
newest firmware too.

Thanks again for your help.

All the best,

Don

**Reply by dnewman-gpsw** [reply]
I will rework the documentation.  The 400x100 is close to the limit, and even on HERO8 it might frame a video frame or too.  The newer firmware is fast still at rendering the image, so size can go up, a bit.  And greyscale images are 50% faster again.

**Comment by don-gopro-user22** [comment]
Hi David,

Thanks for the update.

I will install the newest firmware and assess performance out in the field.

In your experience, what would be the largest logo size I could use,
without affecting performance?

Thanks Don

**Reply by dnewman-gpsw** [reply]
You are adding load to the camera with any hack. So no idea if there is a completely safe threshold for overlays with livestreaming, this is all new for GP2. That is why it is in Labs, the user in running the experiments.  All my testing was around 200x62.  There are performance advantages if the image width is divisible by 8.

**Comment by don-gopro-user22** [comment]
That's helpful David thanks

**Comment by don-gopro-user22** [comment]
Hi David,

I've noticed that the new firmware converts the logo to black and white by
default. I'll revert back to your previous firmware.

Don

**Reply by dnewman-gpsw** [reply]
I just retested with this firmware, and it is working fine here. 
![image](https://user-images.githubusercontent.com/10856084/232869614-5f18c066-0a20-4da0-b196-7ce18a936e0a.png)

Can you share your PNG file?  As it might be a failure in testing the B&W with particular images.

**Comment by don-gopro-user22** [comment]
Sure David, please find logo attached

Don Kelly
PetMinder Portlaoise
www.petminder.ie
086 1628297

**Reply by dnewman-gpsw** [reply]
I don't think that works, replying via email with an attachment (as I don't see it.) You will need to link it via dropbox or similar.  or email to dnewman at GoPro

**Comment by don-gopro-user22** [comment]
Hi David,

You can access my logo here:

https://drive.google.com/file/d/1rFqdTXpSgC7WdzckHOLruDmgHhpBviC3/view?usp=sharing

Don Kelly
PetMinder Portlaoise
www.petminder.ie
086 1628297

**Reply by dnewman-gpsw** [reply]
Found the bug, it is because the only color used in green.  Even a single red or blue pixel will be a workaround.  It will be fixed in a official release.

**Comment by don-gopro-user22** [comment]
Hi David,

I can confirm that changing one of the logo's pixel colour is a workaround.

Over the next couple of weeks, I will test the performance of the
HERO10LabsV15072a
firmware and my logo and get back to you.

Thanks again for all your help, it is much appreciated,

All the best,

Don

Don Kelly
PetMinder Portlaoise
www.petminder.ie
086 1628297

**Reply by dnewman-gpsw** [reply]
Yes, although the issue is green in YUV, has low U and V, you need a color with a high U or V. Red or Blue will work.
